### PR TITLE
Epoch for best model

### LINKED ISSUE #77: Best metrics logger does not log epoch correctly

### PR DESCRIPTION
**Pull request by samedii**
Closing #77 

- Fixed epoch logging
- Moved the parts of the code that are connected so that they are next to each other
- The trigger is also adding the event to the triggered engines
- Changed from `best/validate/metric` to `best-validate/metric` because tb did not handle it well

### CODE REVIEW

**Review by Jim-Holmstroem — APPROVED** [review]

File: workflow/ignite/handlers/best_model_trigger.py [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
 class BestModelTrigger:
-    def __init__(self, engines_to_trigger, metric_to_track):
+    Event = CustomEvents.NEW_BEST_MODEL
+
+    def __init__(self, metric_name, trigger_engines):
         self.best_score = -np.inf
-        self.engines_to_trigger = engines_to_trigger
-        self.metric_to_track = metric_to_track
+        self.trigger_engines = trigger_engines
+        self.metric_name = metric_name
```
Comment (FelixAbrahamsson) [inline workflow/ignite/handlers/best_model_trigger.py:17]: Curious about the name changes. I feel like the old ones are more descriptive and barely longer, so why the change? To someone reading this code for the first time, I don't think `metric_name` will give them as good of a clue on what that variable is used for. In my opinion variable names should be as descriptive as possible without being overly long.

**Comment by FelixAbrahamsson** [comment]
Both guild and tensorboard are able to handle placing scalars under best/validate on my end, what issues are you having?

Edit: Nevermind, I see it now, it's not logging epochs correctly but just saving one value. Good fix :)

File: workflow/ignite/handlers/best_model_trigger.py (hunk 1 above)
Comment (samedii) [reply workflow/ignite/handlers/best_model_trigger.py:17]: Nothing big. I think `metric_name` is more similar to what it looks like in other places. I think the name of the class is enough to tell us what the metric is used for. `BestModelTrigger(metric_name='accuracy').attach(trainer)`

I also think the naming with "to" was something we are not using anywhere else. What do you think about this?

The second argument is very similar though right? Should probably be optional and be something like `extra_triggered_engines`, I'll change that

Comment (samedii) [reply workflow/ignite/handlers/best_model_trigger.py:17]: I also considered changing to the same `model_score_function` that is used elsewhere rather than using a metric

Comment (FelixAbrahamsson) [reply workflow/ignite/handlers/best_model_trigger.py:17]: Sure, I agree that it's more consistent with our other naming. Let's go with this :)

**Review by FelixAbrahamsson — APPROVED** [review]
